src/main/steps/tasklistresponse/allegations-of-harm-and-violence/SafetyConcernsMapper.ts

```diff
@@ -1,72 +1,177 @@
-  const otherConcerns =
-    {
-      c1AkeepingSafeStatement: req.session.userCase.PRL_c1A_keepingSafeStatement,
-      c1AsupervisionAgreementDetails: req.session.userCase.PRL_c1A_supervisionAgreementDetails,
-      c1AagreementOtherWaysDetails: req.session.userCase.PRL_c1A_agreementOtherWaysDetails,
-      c1AotherConcernsDrugs: req.session.userCase.PRL_c1A_otherConcernsDrugs,
-      c1AotherConcernsDrugsDetails: req.session.userCase.PRL_c1A_otherConcernsDrugsDetails,
-      c1AchildSafetyConcerns: req.session.userCase.PRL_c1A_childSafetyConcerns,
-      c1AchildSafetyConcernsDetails: req.session.userCase.PRL_c1A_childSafetyConcernsDetails,
-    } ?? {};
+export const prepareRequest = (respondent: Respondent, req: AppRequest): PRL_C1ASafteyConcerns_total => {
+  const {
+    PRL_c1A_haveSafetyConcerns,
+    PRL_c1A_safetyConernAbout,
+    PRL_c1A_concernAboutChild,
+    PRL_c1A_concernAboutRespondent,
+    PRL_c1A_keepingSafeStatement,
+    PRL_c1A_supervisionAgreementDetails,
+    PRL_c1A_agreementOtherWaysDetails,
+    PRL_c1A_otherConcernsDrugs,
+    PRL_c1A_otherConcernsDrugsDetails,
+    PRL_c1A_childSafetyConcerns,
+    PRL_c1A_childSafetyConcernsDetails,
+    PRL_c1A_abductionReasonOutsideUk,
+    PRL_c1A_childsCurrentLocation,
+    PRL_c1A_childrenMoreThanOnePassport,
+    PRL_c1A_possessionChildrenPassport,
+    PRL_c1A_provideOtherDetails,
+    PRL_c1A_passportOffice,
+    PRL_c1A_abductionPassportOfficeNotified,
+    PRL_c1A_previousAbductionsShortDesc,
+    PRL_c1A_policeOrInvestigatorInvolved,
+    PRL_c1A_policeOrInvestigatorOtherDetails,
+    PRL_c1A_childAbductedBefore,
+    PRL_c1A_safteyConcerns,
+  } = req.session.userCase;
+  let request: PRL_C1ASafteyConcerns_total = {};
 
-  const Abductions =
-    {
-      c1AabductionReasonOutsideUk: req.session.userCase.PRL_c1A_abductionReasonOutsideUk,
-      c1AchildsCurrentLocation: req.session.userCase.PRL_c1A_childsCurrentLocation,
-      c1AchildrenMoreThanOnePassport: req.session.userCase.PRL_c1A_childrenMoreThanOnePassport,
-      c1ApossessionChildrenPassport: req.session.userCase.PRL_c1A_possessionChildrenPassport,
-      c1AprovideOtherDetails: req.session.userCase.PRL_c1A_provideOtherDetails,
-      c1ApassportOffice: req.session.userCase.PRL_c1A_passportOffice,
-      c1AabductionPassportOfficeNotified: req.session.userCase.PRL_c1A_abductionPassportOfficeNotified,
-      c1ApreviousAbductionsShortDesc: req.session.userCase.PRL_c1A_previousAbductionsShortDesc,
-      c1ApoliceOrInvestigatorInvolved: req.session.userCase.PRL_c1A_policeOrInvestigatorInvolved,
-      c1ApoliceOrInvestigatorOtherDetails: req.session.userCase.PRL_c1A_policeOrInvestigatorOtherDetails,
-      c1AchildAbductedBefore: req.session.userCase.PRL_c1A_childAbductedBefore,
-    } ?? {};
-  if (req.session.userCase.PRL_c1A_concernAboutChild) {
-    const safetyConcerns = req.session.userCase.PRL_c1A_safteyConcerns ?? {};
+  if (PRL_c1A_concernAboutChild?.length) {
```

Comment:
If concerns about child is undefined will this not throw a null point exception, since undefined does not have a length property?

Reply:
No it wont. ? prop is added as a defensive check